Avatar: add padding support

colorful-tones opened the pull request:
Related:

- https://github.com/WordPress/gutenberg/issues/43241
- https://github.com/WordPress/gutenberg/issues/43243

## What?

Add padding support to the Avatar block.

## Why?

To create consistency across blocks.

## How?

Added the relevant block support in `block.json`

## Testing Instructions

1. Create a new post
2. Insert a new Avatar.
3. Adding Padding and configure.

Also, test in Global Styles as well.

1. Go to Appearance > Editor (beta)
2. Add an Avatar block to the template
3. Navigate to the Global Styles and choose Avatar block
4. Configure padding and verify

![avatar-block-padding](https://user-images.githubusercontent.com/405912/186162973-6ae17b47-d47c-4927-b01b-5cac22df370b.gif)

<img width="1714" alt="Screen Shot 2022-08-23 at 8 11 02 AM" src="https://user-images.githubusercontent.com/405912/186155116-465f7ba3-291e-4c8e-bfc6-b30092c3f8b0.png">

colorful-tones committed:
Update DateTimePicker readme

colorful-tones committed:
Merge branch 'WordPress:trunk' into trunk

colorful-tones committed:
Merge branch 'WordPress:trunk' into trunk

colorful-tones committed:
Merge branch 'WordPress:trunk' into trunk

colorful-tones committed:
Merge branch 'WordPress:trunk' into trunk

colorful-tones committed:
Merge branch 'WordPress:trunk' into trunk

colorful-tones committed:
Merge branch 'WordPress:trunk' into trunk

colorful-tones committed:
Merge branch 'WordPress:trunk' into trunk

colorful-tones committed:
Merge branch 'WordPress:trunk' into trunk

colorful-tones committed:
Merge branch 'WordPress:trunk' into trunk

colorful-tones committed:
Add padding support for Avatar block

ndiego commented:
@colorful-tones I believe you will want to add `box-sizing: border-box` as well to this. See [this comment](https://github.com/WordPress/gutenberg/pull/43458#pullrequestreview-1079770981) when I added padding to the Query Title. Following the implementation in that PR should do the trick!

colorful-tones committed:
add box-sizing for frontend

colorful-tones commented:
> I believe you will want to add box-sizing: border-box as well to this.

@ndiego good call! I added `box-sizing` to `style.css` and I updated the gif attached in the original description.

ndiego commented on packages/block-library/src/avatar/style.scss:
```diff
@@ -1,4 +1,6 @@
 .wp-block-avatar {
+	box-sizing: border-box;
```
This is a small nit, but can you add the comment 

> // This block has customizable padding, border-box makes that more predictable.

This keeps the [implementation consistent](https://github.com/WordPress/gutenberg/pull/43458/files) with other blocks and makes it easy to understand why this has been added. Otherwise, this PR looks good to go.

colorful-tones replied:
Again, nothing too small, and thanks for the nitpicking! I addressed the missing comment. 👍

colorful-tones committed:
Merge branch 'WordPress:trunk' into trunk

colorful-tones committed:
Merge branch 'trunk' into add/avatar-padding-support

colorful-tones committed:
Additional comment clarification for box-sizing

colorful-tones committed:
Merge branch 'WordPress:trunk' into trunk

carolinan commented:
As with the site logo block, I don't think the top and bottom margins are working?

colorful-tones committed:
Merge branch 'WordPress:trunk' into trunk

colorful-tones committed:
Merge branch 'trunk' into add/avatar-padding-support

colorful-tones commented:
> I don't think the top and bottom margins are working?

@carolinan From all of my tests `margin` and `padding` on all axes are working. 

Here is today's test 

https://user-images.githubusercontent.com/405912/187995982-805ef7fe-da82-445d-a83d-f468ca26f90f.mp4

Here is my testing environment:

- WordPress 6.0.2
- Twenty Twenty-Two 1.2

carolinan reviewed (approved):
With margin set in theme.json only, front view, empty theme:
<img width="1512" alt="Screen Shot 2022-09-02 at 03 47 40" src="https://user-images.githubusercontent.com/7422055/188041548-67fb090f-ed76-4b23-a507-08798e7bce43.png">

Either way, rightly, this PR touches the **padding**, which does work.

andrewserong reviewed (approved):
Thanks for working on this one @colorful-tones, it's working nicely for me, and the padding sits nicely next to an adjacent Group block:

<img width="699" alt="image" src="https://user-images.githubusercontent.com/14988353/188539545-28bfc880-77dc-491c-815b-42f61701c424.png">

The margin issue appears to be the same one as discussed in https://github.com/WordPress/gutenberg/pull/43520#pullrequestreview-1096870277 (and written up in #43404) so not a blocker for merging this one in, I don't think. Update: merged and updated the tracking issue.

LGTM! ✨
